src/adaptations/device-layer/Makefile.am

```diff
@@ -24,44 +24,58 @@ include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 
 if CONFIG_DEVICE_LAYER
 
-lib_LIBRARIES                   		= libDeviceLayer.a
+lib_LIBRARIES                           = libDeviceLayer.a
 
-libDeviceLayer_a_CPPFLAGS        = \
-    -I$(top_srcdir)/src/include    \
-    $(LWIP_CPPFLAGS)               \
-    $(SOCKETS_CPPFLAGS)            \
+libDeviceLayer_a_CPPFLAGS               = \
+    -I$(top_srcdir)/src/include           \
+    $(LWIP_CPPFLAGS)                      \
+    $(SOCKETS_CPPFLAGS)                   \
     $(NULL)
 
-libDeviceLayer_a_SOURCES         = \
-    AESBlockCipher.cpp                  \
-    BLEManager.cpp                      \
-    CASEAuth.cpp                        \
-    ConfigurationManager.cpp            \
-    ConnectivityManager.cpp             \
-    DeviceControlServer.cpp             \
-    DeviceDescriptionServer.cpp         \
-    DeviceIdentityTraitDataSource.cpp   \
-    EchoServer.cpp                      \
-    Entropy.cpp                         \
-    ESPUtils.cpp                        \
-    FabricProvisioningServer.cpp        \
-    Globals.cpp                         \
-    Logging.cpp                         \
-    NetworkInfo.cpp                     \
-    NetworkProvisioningServer.cpp       \
-    PlatformManager.cpp                 \
-    ServiceDirectoryManager.cpp         \
-    ServiceProvisioningServer.cpp       \
-    ServiceTunnelAgent.cpp              \
-    TestDeviceIds.cpp                   \
-    Time.cpp                            \
-    TimeSyncManager.cpp                 \
-    TraitManager.cpp                    \
-    Warm.cpp                            \
+libDeviceLayer_a_SOURCES                = \
+    CASEAuth.cpp                          \
+    DeviceControlServer.cpp               \
+    DeviceDescriptionServer.cpp           \
+    DeviceIdentityTraitDataSource.cpp     \
+    EchoServer.cpp                        \
+    FabricProvisioningServer.cpp          \
+    GeneralUtils.cpp                      \
+    Globals.cpp                           \
+    NetworkInfo.cpp                       \
+    PersistedStorage.cpp                  \
+    ServiceDirectoryManager.cpp           \
+    ServiceProvisioningServer.cpp         \
+    ServiceTunnelAgent.cpp                \
+    SystemEventSupport.cpp                \
+    SystemTimerSupport.cpp                \
+    TestDeviceIds.cpp                     \
+    TimeSyncManager.cpp                   \
+    TraitManager.cpp                      \
     trait-support/weave/trait/description/DeviceIdentityTrait.cpp \
     $(NULL)
 
-EXTRA_DIST                       = \
+if WEAVE_DEVICE_LAYER_TARGET_ESP32
```

Comment:
I suspect it will be more scalable over time if we pick these up through a device-layer/ESP32/Makefile.am or some such. Thoughts?

Reply:
I'm unclear on how that would work.  These source files need to end up in the same library (libDeviceLayer.a), which is built by this Makefile.

Perhaps its possible to include a platform-specific Makefile during the bootstrap step?

Reply:
I was thinking something like:

if WEAVE_DEVICE_LAYER_TARGET_ESP32
include device-layer/ESP32/Makefile.am
endif

or:

if WEAVE_DEVICE_LAYER_TARGET_ESP32
include device-layer/ESP32/ESP32.am
endif

or something like that. We use an approach similar to this in a few other locations in the project.

Reply:
Per our discussion off-line, I will leave the Makefile structure as-is and revisit when there are other platform adaptations.